app/src/main/java/com/android/sample/ui/swipePage/SwipePage.kt

```diff
@@ -89,169 +106,193 @@ fun SwipePage(navigationActions: NavigationActions) {
-          color = MaterialTheme.colorScheme.onSecondary,
-      )
-    }
-    Column(modifier = Modifier.verticalScroll(rememberScrollState())) {
-      // Display with animation Recipe Description
-      AnimatedVisibility(
-          visible = isDescriptionVisible, enter = fadeIn(animationSpec = tween(500))) {
-            Text(
-                modifier = Modifier.testTag("RecipeEntireDescription"),
-                text = getRecipeDescription(),
-                style = MaterialTheme.typography.bodyMedium,
-                color = MaterialTheme.colorScheme.onSecondary,
-            )
-          }
-
-      Spacer(modifier = Modifier.padding(5.dp))
+          // Display Recipe Name
+          RecipeDescription(name, "4.5")
+        }
+    Spacer(modifier = Modifier.padding(2.dp))
+    Row(verticalAlignment = Alignment.CenterVertically) { Tag(tag) }
+  }
+}
 
-      // Display Recipe Timing
-      Row(
-          modifier = Modifier,
-          verticalAlignment =
-              Alignment.CenterVertically // Aligns the icon and text vertically centered
-          ) {
-            Icon(
-                painter = painterResource(R.drawable.timer),
-                contentDescription = "recipes timing",
-                modifier = Modifier.size(24.dp) // Adjust size to match the text size
-                )
+/**
+ * Composable for the Recipe Description
+ *
+ * @param name - Recipe Name
+ * @param rate - Recipe Rate
+ */
+@Composable
+fun RecipeDescription(name: String, rate: String) {
+  Text(
+      modifier = Modifier.testTag("recipeName"),
+      text = name,
+      style = MaterialTheme.typography.bodyLarge,
+      color = MaterialTheme.colorScheme.onSecondary,
+  )
+  Row(
+      horizontalArrangement = Arrangement.End,
+      modifier = Modifier.size(75.dp, 24.dp) // Padding for inside spacing
+      ) {
+        Icon(
+            painter = painterResource(R.drawable.star_rate),
+            contentDescription = "Star rate",
+            modifier = Modifier.size(24.dp).testTag("recipeStar"),
+            tint = starColor)
 
-            Spacer(modifier = Modifier.padding(5.dp))
+        Spacer(modifier = Modifier.padding(5.dp))
 
-            Text(
-                text = "30min",
-                style = MaterialTheme.typography.bodyLarge,
-                fontSize = 22.sp,
-                color = MaterialTheme.colorScheme.onSecondary,
-            )
-          }
-    }
-  }
+        Text(
+            text = rate,
+            modifier = Modifier.testTag("recipeRate"),
+            style = MaterialTheme.typography.bodyLarge,
+            fontSize = 24.sp,
+            color = MaterialTheme.colorScheme.onSecondary,
+        )
+      }
 }
 
-/** @return The recipe description hard coded for now */
+/**
+ * Composable for the Tags of the recipe
+ *
+ * @param tag - Tag Name
+ */
 @Composable
-private fun getRecipeDescription(): String {
-  return hardCodedDesacription
+private fun Tag(tag: String) {
```

Comment:
Same that the other 2 composables : [magic padding](https://stackoverflow.com/questions/41235908/padding-value-in-android) .

Reply:
I made the changes you suggested, similar to the other two composables: I replaced the magic padding values with named constants for better readability and easier future adjustments. Let me know if anything else needs tweaking!